lightning/src/ln/outbound_payment.rs

```diff
@@ -962,10 +992,37 @@ impl OutboundPayments {
 				log_trace!(logger, "Received failure of HTLC with payment_hash {} after payment completion", log_bytes!(payment_hash.0));
 				return
 			}
-			let is_retryable_now = payment.get().is_retryable_now();
+			let is_retryable_now = payment.get().is_auto_retryable_now();
 			if let Some(scid) = short_channel_id {
 				payment.get_mut().insert_previously_failed_scid(scid);
 			}
+
+			// We want to move towards only using the `PaymentParameters` in the outbound payments
+			// map. However, for backwards-compatibility, we still need to support passing the
+			// `PaymentParameters` data that was shoved in the HTLC (and given to us via
```

Comment:
Can we document somewhere (maybe on the serialization) where the `PaymentParameters` shoved in the HTLC has been made redundant, so we can deprecate/remove it in the future?